public/src/overmind/actions.tsx

```diff
@@ -54,36 +55,43 @@ export const receiveSubmission = ({ state }: Context, submission: Submission): v
 
 /** Fetches and stores an authenticated user in state */
 export const getSelf = async ({ state, effects }: Context): Promise<boolean> => {
-    const user = await effects.grpcMan.getUser()
-    if (user.data) {
-        state.self = user.data
-        return true
+    const response = await effects.client.getUser({})
+    if (response.error) {
+        return false
     }
-    return false
+    state.self = response.message
+    return true
 }
 
 /** Gets all enrollments for the current user and stores them in state */
 export const getEnrollmentsByUser = async ({ state, effects }: Context): Promise<void> => {
-    const result = await effects.grpcMan.getEnrollmentsByUser(state.self.ID)
-    if (result.data) {
-        state.enrollments = result.data.enrollments
-        for (const enrollment of state.enrollments) {
-            state.status[enrollment.courseID.toString()] = enrollment.status
+    const response = await effects.client.getEnrollments({
+        FetchMode: {
+            case: "userID",
+            value: state.self.ID,
         }
+    })
+    if (response.error) {
```

Comment:
Should this error be shown to the user somewhere in the frontend?

Reply:
Maybe we could have an error log, but not sure it makes sense to expose such errors to the user (other than us)...